dataplicity/m2m/wsclient.py

```diff
@@ -332,6 +332,7 @@ def on_disconnected(self):
         """Called when ws socket closes."""
         self.identity = None
         self.clear_callbacks()
+        self.hard_close_channels()
```

Comment:
When the websocket disconnects it will close all the channels, which will kill the terminal processes.